frame/nfts/src/features/lock.rs

```diff
@@ -0,0 +1,92 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use frame_support::pallet_prelude::*;
+
+/// Lock functions allow to lock collection/items metadata and attributes.
+/// Additionally, it's possible to make all collection items non-transferable.
+/// Those settings are irreversible.
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	pub fn do_lock_collection(
+		origin: T::AccountId,
+		collection: T::CollectionId,
+		lock_config: CollectionConfig,
+	) -> DispatchResult {
+		let details =
+			Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
+		ensure!(origin == details.freezer, Error::<T, I>::NoPermission);
+
+		CollectionConfigOf::<T, I>::try_mutate(collection, |maybe_config| {
+			let config = maybe_config.as_mut().ok_or(Error::<T, I>::UnknownCollection)?;
```

Comment:
do we need to be able to distinguish this error from the one above? The more specific the error, the less support calls.

Reply:
In a lot of places (except the one you pointed out:) I removed the collection existence check and the first validation will be done over the config. So that if I pass a wrong id, I would get the `UnknownCollection` and not a `NoConfig` which would make more sense give the config would always exists together with collection

Reply:
Though for that particular place that you highlighted I agree it would make sense to change the error message